.github/workflows/build.yml

```diff
@@ -0,0 +1,265 @@
+name: Main build
+
+env:
+  EXPERIMENTAL_JAVA: 17
+
+on:
+  # We want to trigger our builds all the time for the default branch and
+  # for tags starting with `v`.
+  push:
+    branches: 'main'
+    tags-ignore: 
+      - 'v**'
+  # Each pull request is important to us, doesnt matter from which branch.
+  # Furthermore, we do not want to build on just the default GitHub Action
+  # events, we also want to react onto `labeled` events for our extended
+  # build execution.
+  pull_request:
+    branches: '*'
+    types: [labeled, opened, synchronize, reopened]
+  # We also utilize this pipeline for releasing. By providing a `version`
+  # and setting releasing to `true`, we can trigger a release.
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Version string, e.g. 1.4.0'
+        required: true
+      releasing:
+        description: "Is this a release build? true/false"
+        required: true
+        default: "false"
+
+jobs:
+  # Basic Gradle Wrapper validation. This step is run first, due to security
+  # concerns. Running our whole pipeline with gradle with a corrupt wrapper,
+  # could expose our secrets, and we definitely want to prevent this.
+  validation:
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    name: Gradle Wrapper Validation
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@v2
+      - name: Validate Gradle wrapper
+        uses: gradle/wrapper-validation-action@v1
+
+  # Our basic build step as we want the Contributors to have a nice developer
+  # experience with little build time and a lot of information. Therefore we
+  # only build this on our default java version and for all operating systems
+  # we are supporting. This allows for a fast execution and fast feedback, the
+  # extended tests are run later on some conditions.
+  basic:
+    runs-on: ${{ matrix.os }}-latest
+    needs: validation
+    timeout-minutes: 15
+    strategy:
+      matrix:
+        # test against latest update of each major Java version, as well as specific updates of LTS versions:
+        java: [ 8 ]
+        os: [ubuntu, macos, windows]
+    name: with Java ${{ matrix.java }} on ${{ matrix.os }}
```

Comment:
maybe we should just call that `Basic on ${{ matrix.os }}`

Reply:
But then you don't see that Java version "Basic" is

Reply:
Agreed, lets leave the Java version in there.